torchao/prototype/inductor/fx_passes/int8_sdpa_fusion.py

```diff
@@ -0,0 +1,581 @@
+            is_reduced_type=False,
+            has_convert=True,
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=False,
+            is_batch_size_1=True,
+            is_reduced_type=False,
+            has_convert=False,
+        )
+    )
+
+
+def _register_int8_sdpa_fp32_mask_bs1_lowering():
+    # dtype = float32, with attention mask, batch size == 1
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True, is_batch_size_1=True, is_reduced_type=False, has_convert=True
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True,
+            is_batch_size_1=True,
+            is_reduced_type=False,
+            has_convert=False,
+        )
+    )
+
+
+def _register_int8_sdpa_bf16_lowering():
+    # dtype = bfloat16, without attention mask, batch size > 1
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=False,
+            is_batch_size_1=False,
+            is_reduced_type=True,
+            has_convert=True,
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=False,
+            is_batch_size_1=False,
+            is_reduced_type=True,
+            has_convert=False,
+        )
+    )
+
+
+def _register_int8_sdpa_bf16_mask_lowering():
+    # dtype = bfloat16, with attention mask, batch size > 1
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True, is_batch_size_1=False, is_reduced_type=True, has_convert=True
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True,
+            is_batch_size_1=False,
+            is_reduced_type=True,
+            has_convert=False,
+        )
+    )
+
+
+def _register_int8_sdpa_bf16_bs1_lowering():
+    # dtype = bfloat16, without attention mask, batch size == 1
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=False, is_batch_size_1=True, is_reduced_type=True, has_convert=True
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=False,
+            is_batch_size_1=True,
+            is_reduced_type=True,
+            has_convert=False,
+        )
+    )
+
+
+def _register_int8_sdpa_bf16_mask_bs1_lowering():
+    # dtype = bfloat16, with attention mask, batch size == 1
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True, is_batch_size_1=True, is_reduced_type=True, has_convert=True
+        )
+    )
+    _register_int8_sdpa_pattern(
+        _get_int8_sdpa_final_pattern(
+            has_mask=True, is_batch_size_1=True, is_reduced_type=True, has_convert=False
+        )
+    )
+
+
+def _register_int8_sdpa_lowerings():
+    _register_int8_sdpa_fp32_lowering()
```

Comment:
instead of register it one by one, can we put them into arguments of a register function?

Reply:
Thanks, modified!